tools/python/boutiques/localExec.py

```diff
@@ -206,22 +206,52 @@ def execute(self, mount_strings):
                              for m in mount_strings]
             mount_strings.append(op.realpath('./') + ':' + launchDir)
             if conType == 'docker':
+                envString = " "
+                if envVars:
+                    for (key, val) in list(envVars.items()):
+                        envString += " -e {0}='{1}' ".format(key, val)
                 # export mounts to docker string
                 docker_mounts = " -v ".join(m for m in mount_strings)
                 dcmd = ('docker run --entrypoint=/bin/sh --rm' + envString +
                         ' -v ' + docker_mounts + ' -w ' + launchDir + ' ' +
                         str(conImage) + ' ' + dsname)
             elif conType == 'singularity':
-                singularity_mounts = " -B ".join(m for m in mount_strings)
-                # TODO: Test singularity runtime on cluster
-                dcmd = ('singularity exec' + envString + ' -B ' +
+                envString = ""
+                if envVars:
+                    for (key, val) in list(envVars.items()):
+                        envString += "SINGULARITYENV_{0}='{1}' ".format(key,
+                                                                        val)
+                # TODO: Singularity 2.4.6 default configuration binds: /proc,
+                # /sys, /dev, ${HOME}, /tmp, /var/tmp, /etc/localtime, and
+                # /etc/hosts. This means that any path down-stream shouldn't
+                # be bound on the command-line, as this will currently raise
+                # an exception. See:
+                #   https://github.com/singularityware/singularity/issues/1469
+                #
+                # Previous bind string:
+                #   singularity_mounts = " -B ".join(m for m in mount_strings)
+
+                def_mounts = ["/proc", "/sys", "/dev", "/tmp", "/var/tmp",
+                              "/etc/localtime", "/etc/hosts",
+                              op.realpath(op.expanduser('~')),
+                              op.expanduser('~')]
+                singularity_mounts = ""
+                for m in mount_strings:
+                    if not any(d in m for d in def_mounts):
+                        singularity_mounts += "-B {0} ".format(m)
+                print(mount_strings)
+                print(singularity_mounts)
```

Comment:
Prints should be done only in `bosh`, when using the CLI. We might want to add debug info to the tuple returned by this method if this is useful to the API. Or do this only in debug mode.